src/lib.rs

```diff
@@ -3817,6 +3817,17 @@ mod tests {
     fn filter() {
         use crate::{DefaultExpected, LabelError};
 
+        let parser = any::<_, extra::Err<Rich<_>>>().filter(|_| false);
```

Comment:
Could you change this to `just("a")`? I think that would give us more assurance that it's the `Filter` that's generating this error and not the `any` (helps guard against future refactoring bugs).